### PR TITLE
Cleanup CI

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

# What changes are included in this PR?

* Consistently call clippy with `-D warnings`
* Consistently use setup-builder action
* Consistently test wasm32 as a separate job
* Inline single-line run commands
* Default rust-version in setup-builder
* Add target option to setup-builder

# Are there any user-facing changes?

No

### CODE REVIEW

**Comment by alamb** [comment]
This PR appears to have conflicts in need of addressing

**Comment by tustvold** [comment]
Fixed merge conflict, PTAL

File: .github/actions/setup-builder/action.yaml [hunk 1]
```diff
@@ -51,7 +55,11 @@ runs:
       shell: bash
       run: |
         echo "Installing ${{ inputs.rust-version }}"
-        rustup toolchain install ${{ inputs.rust-version }}
+        rustup toolchain install ${{ inputs.rust-version }} --target ${{ inputs.target }}
         rustup default ${{ inputs.rust-version }}
-        rustup target add wasm32-unknown-unknown
-        echo "CARGO_TARGET_DIR=/github/home/target" >> $GITHUB_ENV
+    - name: Disable debuginfo generation
```
Comment (alamb) [inline .github/actions/setup-builder/action.yaml:60]:
```suggestion
    - name: Disable debuginfo generation
    # Disable full debug symbol generation to speed up CI build and keep memory down
    # "1" means line tables only, which is useful for panic tracebacks.    
```

I think the rationale is important so it doesn't get lost accidentally along the way

File: .github/workflows/parquet.yml [hunk 2]
```diff
@@ -93,35 +79,42 @@ jobs:
...
+        run: cargo check -p parquet --all-targets --all-features
       - name: Check compilation  --all-targets --no-default-features --features json
-        run: |
-          cargo check -p parquet --all-targets --no-default-features --features json
-      - name: Check compilation wasm32-unknown-unknown
-        run: |
-          cargo check -p parquet --no-default-features --features cli,snap,flate2,brotli --target wasm32-unknown-unknown
+        run: cargo check -p parquet --all-targets --no-default-features --features json
+
+  # test the parquet crate builds against wasm32 in stable rust
```
Comment (alamb) [inline .github/workflows/parquet.yml:100]: this is a new check, right?

**Review by alamb — APPROVED** [review]
LGTM

File: .github/workflows/parquet.yml (hunk 2 above)
Comment (tustvold) [reply .github/workflows/parquet.yml:100]: Yup - follow on to #2896

**Pull request merged**